sparse/array.go

```diff
@@ -12,6 +12,7 @@ func UnsafeArray(base unsafe.Pointer, length int, offset uintptr) Array {
 }
 
 func (a Array) Len() int                   { return int(a.len) }
+func (a Array) Nil() bool                  { return a.ptr == nil }
```

Comment:
Is this change related to the original issue?

Reply:
The nil check is required [here](https://github.com/parquet-go/parquet-go/blob/acef7b2f0ddfe6ea1cfdf626369adb63d8e42ec4/column_buffer.go#L2205) to make sure the definition level is increased only for non nil lists